obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+    f.seek(0)
+    for line in f:
+        if line[79] in [' ', '1']:
+            topline = line
+            break
+        if line[79] == '7':
+            f.close()
+            raise IOError('No header found, corrupt s-file?')
+    try:
+        sfile_seconds = int(topline[16:18])
+        if sfile_seconds == 60:
+            sfile_seconds = 0
+            add_seconds = 60
+        else:
+            add_seconds = 0
+        new_event.origins.append(Origin())
+        new_event.origins[0].time = UTCDateTime(int(topline[1:5]),
+                                                int(topline[6:8]),
+                                                int(topline[8:10]),
+                                                int(topline[11:13]),
+                                                int(topline[13:15]),
+                                                sfile_seconds,
+                                                int(topline[19:20]) *
+                                                100000)\
+            + add_seconds
+    except:
+        f.close()
+        IOError("Couldn't read a date from sfile")
+    # new_event.loc_mod_ind=topline[20]
+    new_event.event_descriptions.append(EventDescription())
+    new_event.event_descriptions[0].text = topline[21:23]
+    # new_event.ev_id=topline[22]
+    if not _float_conv(topline[23:30]) == 999:
+        new_event.origins[0].latitude = _float_conv(topline[23:30])
+        new_event.origins[0].longitude = _float_conv(topline[31:38])
+        new_event.origins[0].depth = _float_conv(topline[39:43]) * 1000
+    else:
+        # The origin 'requires' a lat & long
+        new_event.origins[0].latitude = float('NaN')
+        new_event.origins[0].longitude = float('NaN')
+        new_event.origins[0].depth = float('NaN')
+    # new_event.depth_ind = topline[44]
+    # new_event.loc_ind = topline[45]
+    new_event.creation_info = CreationInfo(agency_id=topline[45:48].
+                                           strip())
+    ksta = Comment(text='Number of stations=' +
+                   topline[49:51].strip())
+    new_event.origins[0].comments.append(ksta)
+    # new_event.origins[0].nsta??? = _int_conv(topline[49:51])
+    if not _float_conv(topline[51:55]) == 999:
+        new_event.origins[0].time_errors['Time_Residual_RMS'] = \
+            _float_conv(topline[51:55])
+    # Read in magnitudes if they are there.
+    if len(topline[59].strip()) > 0:
+        new_event.magnitudes.append(Magnitude())
+        new_event.magnitudes[0].mag = _float_conv(topline[56:59])
+        new_event.magnitudes[0].magnitude_type = topline[59]
+        new_event.magnitudes[0].creation_info = \
+            CreationInfo(agency_id=topline[60:63].strip())
+        new_event.magnitudes[0].origin_id = new_event.origins[0].\
+            resource_id
+    if len(topline[67].strip()) > 0:
+        new_event.magnitudes.append(Magnitude())
+        new_event.magnitudes[1].mag = _float_conv(topline[64:67])
```

Comment:
You don't know that the previous condition was true, so you don't know that `1` is the correct index into the list.